glib/src/collections/strv.rs

```diff
@@ -443,11 +446,11 @@ impl StrV {
         if len == 0 {
             StrV::default()
         } else {
-            // Need to fully copy the array here.
-            let slice = Self::from_glib_borrow_num(ptr, len);
-            let mut s = Self::new();
-            s.extend_from_slice(slice);
-            s
+            StrV {
+                ptr: ptr::NonNull::new_unchecked(ffi::g_strdupv(ptr as *mut _)),
```

Comment:
The reason for the previous implementation was that we know the length while `g_strdupv()` first needs to determine the length before copying

Reply:
Yes, but `extend_from_slice` requires `AsRef` so it cannot be used anymore, I guess I can re-implement it

Reply:
Ah I see. Well, worst case start with `g_memdup()` :)

Reply:
I need to copy all the strings as well, so memdup does not sound the right thing... I simply allocated and then copied all the strings